keyboards/yandrstudio/bl75/bl75.h

```diff
@@ -0,0 +1,74 @@
+/* Copyright 2022 JasonRen(biu)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
+#include "quantum.h"
+
+#define LAYOUT( \
+    K000, K001, K002, K003, K004, K005, K006, K007, K008, K009, K010, K011, K012, K013, K014, \
+    K100, K101, K102, K103, K104, K105, K106, K107, K108, K109, K110, K111, K112, K113, K114, \
+    K200, K201, K202, K203, K204, K205, K206, K207, K208, K209, K210, K211, K212, K213, K214, \
+    K300, K301, K302, K303, K304, K305, K306, K307, K308, K309, K310, K311,       K313, K314, \
+    K400,       K402, K403, K404, K405, K406, K407, K408, K409, K410, K411, K412, K413, K414, \
+    K500, K501, K502,                   K506,                   K510, K511, K512, K513, K514 \
+    )\
+    { \
+    { K000,  K001,  K002,  K003,  K004,  K005,  K006,  K007,  K008,  K009,  K010,  K011,  K012,  K013,  K014  }, \
+    { K100,  K101,  K102,  K103,  K104,  K105,  K106,  K107,  K108,  K109,  K110,  K111,  K112,  K113,  K114  }, \
+    { K200,  K201,  K202,  K203,  K204,  K205,  K206,  K207,  K208,  K209,  K210,  K211,  K212,  K213,  K214  }, \
+    { K300,  K301,  K302,  K303,  K304,  K305,  K306,  K307,  K308,  K309,  K310,  K311,  KC_NO, K313,  K314  }, \
+    { K400,  KC_NO, K402,  K403,  K404,  K405,  K406,  K407,  K408,  K409,  K410,  K411,  K412,  K413,  K414  }, \
+    { K500,  K501,  K502,  KC_NO, KC_NO, KC_NO, K506,  KC_NO, KC_NO, KC_NO, K510,  K511,  K512,  K513,  K514  }  \
+}
+
+enum keyboard_keycodes {
+#ifdef VIA_ENABLE
+    LOCK_GUI = USER00,
+#ifdef RGB_MATRIX_ENABLE
+    KC_KEY_UNDER_RGBSW,
+#endif
+    TOG_MACOS_KEYMAP,
+    KC_MISSION_CONTROL,
+    KC_LAUNCHPAD,
+    NEW_SAFE_RANGE = SAFE_RANGE  // Important!
+#else
+    LOCK_GUI = SAFE_RANGE,
+#ifdef RGB_MATRIX_ENABLE
+    KC_KEY_UNDER_RGBSW,
+#endif
+    TOG_MACOS_KEYMAP,
+    KC_MISSION_CONTROL,
+    KC_LAUNCHPAD,
+    NEW_SAFE_RANGE   // Important!
+#endif
+};
```

Comment:
This is a lot cleaner.
```suggestion
enum keyboard_keycodes {
#ifdef VIA_ENABLE
    LOCK_GUI = USER00,
#else
    LOCK_GUI = SAFE_RANGE,
#endif
    KC_KEY_UNDER_RGBSW,
    TOG_MACOS_KEYMAP,
    KC_MISSION_CONTROL,
    KC_LAUNCHPAD,
#ifndef VIA_ENABLE
    NEW_SAFE_RANGE   // Important!
#else
#    define NEW_SAFE_RANGE SAFE_RANGE
#endif
};
```

Also, shouldn't hide keycodes behind defines,.  It can/will cause issues. Like if via is enabled but rgb matrix wasn't, it will shift around the user keycodes.